### ISSUE TITLE
\ -f "file" dumps core

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#4620]
Migrated from [rt.perl.org#7947](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=7947) (status was 'resolved')

Searchable as RT7947$

**Comment by p5pRT** [comment]
### From joe@sunstarsys.com



&nbsp;        % perl \-wle ' \\\-f "blah" '
&nbsp;        Useless use of single ref constructor in void context at \-e line 1\.
&nbsp;        Segmentation fault \(core dumped\)

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=low

Site configuration information for perl v5.6.1:

Configured by joe at Tue Jun 12 14:52:01 EDT 2001.

Summary of my perl5 (revision 5.0 version 6 subversion 1) configuration:
  Platform:
    osname=linux, osvers=2.4.3, archname=i586-linux-thread-multi
    uname='linux mumonkan.sunstarsys.com 2.4.3 #3 thu apr 26 11:39:34 edt 2001 i586 unknown '
    config_args=''
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=define use5005threads=undef useithreads=define usemultiplicity=define
    useperlio=define d_sfio=define uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
  Compiler:
    cc='cc', ccflags ='-D_REENTRANT -DDEBUGGING -fno-strict-aliasing -I/usr/local/include -I/usr/gnu/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O2 -g',
    cppflags='-D_REENTRANT -DDEBUGGING -fno-strict-aliasing -I/usr/local/include -I/usr/gnu/include'
    ccversion='', gccversion='2.96 20000731 (Red Hat Linux 7.0)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, usemymalloc=y, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -L/usr/local/lib -L/usr/gnu/lib'
    libpth=/usr/local/lib /usr/gnu/lib /lib /usr/lib
    libs=-lsfio -lnsl -lndbm -lgdbm -ldl -lm -lpthread -lc -lposix -lcrypt -lutil
    perllibs=-lsfio -lnsl -ldl -lm -lpthread -lc -lposix -lcrypt -lutil
    libc=/lib/libc-2.1.92.so, so=so, useshrplib=false, libperl=libperl.a
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic'
    cccdlflags='-fpic', lddlflags='-shared -L/usr/local/lib -L/usr/gnu/lib'

Locally applied patches:
    


@INC for perl v5.6.1:
    /usr/lib/perl5/5.6.1/i586-linux-thread-multi
    /usr/lib/perl5/5.6.1
    /usr/lib/perl5/site_perl/5.6.1/i586-linux-thread-multi
    /usr/lib/perl5/site_perl/5.6.1
    /usr/lib/perl5/site_perl
    /usr/local/lib/perl5/vendor/lib/perl5/vendor_perl/5.6.1/i586-linux-thread-multi
    /usr/local/lib/perl5/vendor/lib/perl5/vendor_perl/5.6.1
    /usr/local/lib/perl5/vendor/lib/perl5/vendor_perl
    .


Environment for perl v5.6.1:
    HOME=/home/joe
    LANG=en_US
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/usr/kerberos/sbin:/home/joe/bin:/home/joe/bin:/usr/kerberos/bin:/usr/bin:/bin:/usr/X11R6/bin:/usr/local/bin:/opt/bin:/usr/gnu/bin:/usr/X11R6/bin:/usr/bin/mh:/usr/gnu/bin:/usr/X11R6/bin:/usr/bin/mh:/usr/bin/mh:/usr/bin/mh:/usr/bin/mh
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @schwern

I can get it to core dump with a little more effort&#8203;:

$ perl \-wle '$foo = \\\-f "blah";  print $foo;  print $$foo'  
\-f on unopened filehandle at \-e line 1\.
SCALAR\(0x100129a0\)
Use of uninitialized value in print at \-e line 1\.

Segmentation fault

Interestingly enough\, the filename in question has to be less then 12
characters\.  Maybe something to do with the 8\.3 filehandle handling code\.

$ perl \-wle '$foo = \\\-f "blahblahblax";  print $foo;  print $$foo'
\-f on unopened filehandle at \-e line 1\.
SCALAR\(0x100129a0\)
Use of uninitialized value in print at \-e line 1\.


but it works under bleadperl

$ bleadperl \-wle '$foo = \\\-f "blah";  print $foo;  print $$foo'
SCALAR\(0x101947d8\)
Use of uninitialized value in print at \-e line 1\.

$ bleadperl \-wle '$foo = \\\-f "Mailbox";  print $foo;  print $$foo'
SCALAR\(0x10194cb0\)
1


Grab the latest bleadperl snapshot and see if everything works ok on
your system & prefered config\.
ftp&#8203;://ftp\.funet\.fi/pub/languages/perl/snap/


Meanwhile\, here's a test patch&#8203;:

\-\-\- t/run/kill\_perl\.t    2001/11/27 20&#8203;:20&#8203;:58    1\.1
\+\+\+ t/run/kill\_perl\.t    2001/11/27 20&#8203;:29&#8203;:09
@&#8203;@&#8203; \-809\,3 \+809\,9 @&#8203;@&#8203;
 EXPECT
 Can't modify constant item in read at \- line 1\, near "1\)"
 Execution of \- aborted due to compilation errors\.
\+\#\#\#\#\#\#\#\# scalar ref to file test operator segfaults on 5\.6\.1 \[ID 20011127\.155\]
\+\# This only happens if the filename is 11 characters or less\.
\+$foo = \\\-f "blah";
\+print "ok" if ref $foo && \!$$foo;
\+EXPECT
\+ok



\-\- 

Michael G\. Schwern   \<schwern@&#8203;pobox\.com>    http&#8203;://www\.pobox\.com/~schwern/
Perl Quality Assurance        \<perl\-qa@&#8203;perl\.org>           Kwalitee Is Job One
"Hey kids\!  I'm Beefy the Elf\!    Follow me to Meattart Land\!"
"It's like chewing on a lemon cow\!"
"I like Meattarts 'cause they're meaty\!"
"I like Meattarts 'cause\.\.\. wait\.  These suck\."
&nbsp;    http&#8203;://www\.goats\.com/archive/000608\.html

**Comment by p5pRT** [comment]
### From @schwern

\-\- 

Michael G\. Schwern   \<schwern@&#8203;pobox\.com>    http&#8203;://www\.pobox\.com/~schwern/
Perl Quality Assurance        \<perl\-qa@&#8203;perl\.org>           Kwalitee Is Job One
Good tidings\, my native American Indian friend\!  America will soon again
be yours\!  Please accept 5th Avenue as an initial return\!

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

> This is a bug report for perl from joe@&#8203;sunstarsys\.com\,
> generated with the help of perlbug 1\.33 running under perl v5\.6\.1\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please enter your report here\]
> 
>         % perl \-wle ' \\\-f "blah" '
>         Useless use of single ref constructor in void context at \-e line 1\.
>         Segmentation fault \(core dumped\)

Hopefully some extra machines being in on testing this will shed some
light\.

I get some 5\.6\.0 systems to act the same\, while others do not\.
Most of the systems mentioned have perl 5\.6\.x versions that are
from the source from CPAN\. The Win98 box has AS 623 direct from
ActiveState\, and I think one or two boxes may have RPM packages\,
notably the Mandrake box which would already have shipped with
5\.6\.0 in that version of the OS packaging\.

First\, I can get this down to the following&#8203;:

&nbsp;    perl \-we '\\\-f'

It only dumps core with warnings turned on\, and the dash or something
similar is required\, because

&nbsp;    perl \-we '\\\[\]'

does not trigger the segfault\. OTOH\, this does dump core&#8203;:

&nbsp;    perl \-we 'my $h; $h = \\\-f'

but does not give the 'in void context' warning\.

Testing 5\.005\_03 on all systems wher eI still have it gives&#8203;:

&nbsp;    Stat on unopened file \<> at \-e line 1\.
&nbsp;    Attempt to free unreferenced scalar\.

for all cases which cause the 5\.6\.x series to dump core on RedHat\.

5\.6\.1 that doesn't crash gives this&#8203;:

&nbsp;    Useless use of single ref constructor in void context at \-e line 1\.
&nbsp;    \-f on unopened filehandle at \-e line 1\.
&nbsp;    Attempt to free unreferenced scalar\.

or this&#8203;:

&nbsp;    \-f on unopened filehandle at \-e line 1\.
&nbsp;    Attempt to free unreferenced scalar\.


for the versions without and with the assignment\. 5\.005\_03 seems to
give the same results on both systems\.

Mandrake 8\.0 with perl 5\.6\.0 seems to have the problem\. I believe
this is the original Mandrake perl RPM in this case\, but I may be
mistaken\)\. I tested some of my companys RedHat boxes\, trying to
get a good variety\, but I did not test them all \(since that would
take quite a while\)\.

Here's a summary&#8203;:



Mandrake 8\.0 \-\- segfaults

&nbsp;    $ perl \-v
&nbsp;    This is perl\, v5\.6\.0 built for i386\-linux

&nbsp;    Copyright 1987\-2000\, Larry Wall


&nbsp;    $ uname \-a
&nbsp;    Linux larry\.motion\.net 2\.4\.3\-20mdk \#1 Sun Apr 15 23&#8203;:03&#8203;:10
&nbsp;    CEST 2001 i586 unknown

\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#


RedHat 6\.2 \-\- does not segfault

&nbsp;    $ perl \-v
&nbsp;    This is perl\, v5\.6\.1 built for i686\-linux

&nbsp;    Copyright 1987\-2001\, Larry Wall


&nbsp;    $ uname \-a
&nbsp;    Linux cgi\.motion\.net 2\.2\.14\-5\.0 \#1 Tue Mar 7 21&#8203;:07&#8203;:39
&nbsp;    EST 2000 i686 unknown

\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#


RedHat 6\.2 \(updated kernel\) \-\- segfaults

&nbsp;    $ perl \-v
&nbsp;    This is perl\, v5\.6\.0 built for i686\-linux

&nbsp;    Copyright 1987\-2000\, Larry Wall


&nbsp;    $ uname \-a
&nbsp;    Linux velma 2\.2\.17\-14 \#5 Fri Mar 9 06&#8203;:53&#8203;:54
&nbsp;    CST 2001 i686 unknown

\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#

RedHat 7\.1 \-\- does not segfault

&nbsp;    $ perl \-v
&nbsp;    This is perl\, v5\.6\.0 built for i386\-linux

&nbsp;    Copyright 1987\-2000\, Larry Wall

&nbsp;    $ uname \-a
&nbsp;    Linux mail\.ctnet\.net 2\.4\.2\-2smp \#1 SMP Sun Apr 8 20&#8203;:21&#8203;:34
&nbsp;    EDT 2001 i686 unknown

\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#


RedHat 7\.1 \(updated kernel\) \-\- does not segfault

&nbsp;    $ perl \-v
&nbsp;    This is perl\, v5\.6\.1 built for i686\-linux

&nbsp;    Copyright 1987\-2001\, Larry Wall

&nbsp;    $ uname \-a
&nbsp;    Linux testing\-chris\.motion\.net 2\.4\.12 \#3 SMP Fri Oct 26 10&#8203;:52&#8203;:56
&nbsp;    CDT 2001 i686 unknown

\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#

Windows 98 SE \-\- segfaults

&nbsp;    C&#8203;:\\WINDOWS>perl \-v
&nbsp;    This is perl\, v5\.6\.0 built for MSWin32\-x86\-multi\-thread
&nbsp;    \(with 1 registered patch\, see perl \-V for more detail\)

&nbsp;    Copyright 1987\-2000\, Larry Wall

&nbsp;    Binary build 623 provided by ActiveState Tool Corp\.
&nbsp;    http&#8203;://www\.ActiveState\.com
&nbsp;    Built 16&#8203;:27&#8203;:07 Dec 15 2000


&nbsp;    C&#8203;:\\WINDOWS>ver

&nbsp;    Windows 98 \[Version 4\.10\.2222\]


Here are the long version info blocks for the versions that segfault&#8203;:


Mandrake 8\.0\, perl 5\.6\.0&#8203;:
\#\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-

Summary of my perl5 \(revision 5\.0 version 6 subversion 0\) configuration&#8203;:
&nbsp;  Platform&#8203;:
&nbsp;    osname=linux\, osvers=2\.4\.1\-20mdksmp\, archname=i386\-linux
&nbsp;    uname='linux bi\.mandrakesoft\.com 2\.4\.1\-20mdksmp \#1 smp tue feb 20 19&#8203;:21&#8203;:17 cet 2001 i686 unknown '
&nbsp;    config\_args='\-des \-Darchname=i386\-linux \-Dd\_dosuid \-Ud\_csh \-Duseshrplib \-Doptimize=\-O3 \-fomit\-frame\-pointer \-pipe \-mcpu=pentiumpro \-march=i586 \-ffast\-math \-Dprefix=/usr \-Di\_ndbm \-Di\_gdbm \-Di\_shadow \-Di\_syslog \-Uuselargefiles \-Dman1dir=/usr/share/man/man1 \-Dman3dir=/usr/lib/perl5/man/man3'
&nbsp;    hint=recommended\, useposix=true\, d\_sigaction=define
&nbsp;    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
&nbsp;    useperlio=undef d\_sfio=undef uselargefiles=undef
&nbsp;    use64bitint=undef use64bitall=undef uselongdouble=undef usesocks=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='cc'\, optimize='\-O3 \-fomit\-frame\-pointer \-pipe \-mcpu=pentiumpro \-march=i586 \-ffast\-math'\, gccversion=2\.96 20000731 \(Linux\-Mandrake 8\.0 2\.96\-0\.47mdk\)
&nbsp;    cppflags='\-fno\-strict\-aliasing'
&nbsp;    ccflags ='\-fno\-strict\-aliasing'
&nbsp;    stdchar='char'\, d\_stdstdio=define\, usevfork=false
&nbsp;    intsize=4\, longsize=4\, ptrsize=4\, doublesize=8
&nbsp;    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=12
&nbsp;    ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\, lseeksize=4
&nbsp;    alignbytes=4\, usemymalloc=n\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='cc'\, ldflags =' \-L/usr/local/lib'
&nbsp;    libpth=/usr/local/lib /lib /usr/lib
&nbsp;    libs=\-lnsl \-ldl \-lm \-lc \-lcrypt
&nbsp;    libc=/lib/libc\-2\.2\.2\.so\, so=so\, useshrplib=true\, libperl=libperl\.so
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_dlopen\.xs\, dlext=so\, d\_dlsymun=undef\, ccdlflags='\-rdynamic \-Wl\,\-rpath\,/usr/lib/perl5/5\.6\.0/i386\-linux/CORE'
&nbsp;    cccdlflags='\-fpic'\, lddlflags='\-shared \-L/usr/local/lib'

Characteristics of this binary \(from libperl\)&#8203;:
&nbsp;  Compile\-time options&#8203;:
&nbsp;  Built under linux
&nbsp;  Compiled at Apr  4 2001 18&#8203;:57&#8203;:55
&nbsp;  @&#8203;INC&#8203;:
&nbsp;    /usr/lib/perl5/5\.6\.0/i386\-linux
&nbsp;    /usr/lib/perl5/5\.6\.0
&nbsp;    /usr/lib/perl5/site\_perl/5\.6\.0/i386\-linux
&nbsp;    /usr/lib/perl5/site\_perl/5\.6\.0
&nbsp;    /usr/lib/perl5/site\_perl
&nbsp;    \.

\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#


RedHat 6\.2 \(kernel updated\)\, perl 5\.6\.0&#8203;:
\#\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-

Summary of my perl5 \(revision 5\.0 version 6 subversion 0\) configuration&#8203;:
&nbsp;  Platform&#8203;:
&nbsp;    osname=linux\, osvers=2\.2\.14\-5\.0\, archname=i686\-linux
&nbsp;    uname='linux velma 2\.2\.14\-5\.0 \#1 tue mar 7 21&#8203;:07&#8203;:39 est 2000 i686 unknown '
&nbsp;    config\_args=''
&nbsp;    hint=recommended\, useposix=true\, d\_sigaction=define
&nbsp;    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
&nbsp;    useperlio=undef d\_sfio=undef uselargefiles=define
&nbsp;    use64bitint=undef use64bitall=undef uselongdouble=define usesocks=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='cc'\, optimize='\-O2'\, gccversion=egcs\-2\.91\.66 19990314/Linux \(egcs\-1\.1\.2 release\)
&nbsp;    cppflags='\-fno\-strict\-aliasing'
&nbsp;    ccflags ='\-fno\-strict\-aliasing \-D\_LARGEFILE\_SOURCE \-D\_FILE\_OFFSET\_BITS=64'
&nbsp;    stdchar='char'\, d\_stdstdio=define\, usevfork=false
&nbsp;    intsize=4\, longsize=4\, ptrsize=4\, doublesize=8
&nbsp;    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=12
&nbsp;    ivtype='long'\, ivsize=4\, nvtype='long double'\, nvsize=12\, Off\_t='off\_t'\, lseeksize=8
&nbsp;    alignbytes=4\, usemymalloc=y\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='cc'\, ldflags =' \-L/usr/local/lib'
&nbsp;    libpth=/usr/local/lib /lib /usr/lib
&nbsp;    libs=\-lnsl \-lndbm \-lgdbm \-ldb \-ldl \-lm \-lc \-lposix \-lcrypt
&nbsp;    libc=/lib/libc\-2\.1\.3\.so\, so=so\, useshrplib=false\, libperl=libperl\.a
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_dlopen\.xs\, dlext=so\, d\_dlsymun=undef\, ccdlflags='\-rdynamic'
&nbsp;    cccdlflags='\-fpic'\, lddlflags='\-shared \-L/usr/local/lib'


Characteristics of this binary \(from libperl\)&#8203;:
&nbsp;  Compile\-time options&#8203;: USE\_LONG\_DOUBLE USE\_LARGE\_FILES
&nbsp;  Built under linux
&nbsp;  Compiled at Feb 21 2001 22&#8203;:00&#8203;:13
&nbsp;  @&#8203;INC&#8203;:
&nbsp;    /usr/local/lib/perl5/5\.6\.0/i686\-linux
&nbsp;    /usr/local/lib/perl5/5\.6\.0
&nbsp;    /usr/local/lib/perl5/site\_perl/5\.6\.0/i686\-linux
&nbsp;    /usr/local/lib/perl5/site\_perl/5\.6\.0
&nbsp;    /usr/local/lib/perl5/site\_perl
&nbsp;    \.
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#
\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#


Win98SE\, perl 5\.6\.0 \(ActiveState build 623\)&#8203;:
\#\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
Summary of my perl5 \(revision 5 version 6 subversion 0\) configuration&#8203;:
&nbsp;  Platform&#8203;:
&nbsp;    osname=MSWin32\, osvers=4\.0\, archname=MSWin32\-x86\-multi\-thread
&nbsp;    uname=''
&nbsp;    config\_args='undef'
&nbsp;    hint=recommended\, useposix=true\, d\_sigaction=undef
&nbsp;    usethreads=undef use5005threads=undef useithreads=define usemultiplicity=def
ine
&nbsp;    useperlio=undef d\_sfio=undef uselargefiles=undef
&nbsp;    use64bitint=undef use64bitall=undef uselongdouble=undef usesocks=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='cl'\, optimize='\-O1 \-MD \-DNDEBUG'\, gccversion=
&nbsp;    cppflags='\-DWIN32'
&nbsp;    ccflags ='\-O1 \-MD \-DNDEBUG \-DWIN32 \-D\_CONSOLE \-DNO\_STRICT \-DHAVE\_DES\_FCRYPT
 \-DPERL\_IMPLICIT\_CONTEXT \-DPERL\_IMPLICIT\_SYS \-DPERL\_MSVCRT\_READFIX'
&nbsp;    stdchar='char'\, d\_stdstdio=define\, usevfork=false
&nbsp;    intsize=4\, longsize=4\, ptrsize=4\, doublesize=8
&nbsp;    d\_longlong=undef\, longlongsize=8\, d\_longdbl=define\, longdblsize=10
&nbsp;    ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\, lseeksize
=4
&nbsp;    alignbytes=8\, usemymalloc=n\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='link'\, ldflags ='\-nologo \-nodefaultlib \-release  \-libpath&#8203;:"C&#8203;:\\Perl\\lib\\C
ORE"  \-machine&#8203;:x86'
&nbsp;    libpth="C&#8203;:\\Perl\\lib\\CORE"
&nbsp;    libs=  oldnames\.lib kernel32\.lib user32\.lib gdi32\.lib winspool\.lib  comdlg32
\.\.lib advapi32\.lib shell32\.lib ole32\.lib oleaut32\.lib  netapi32\.lib uuid\.lib wsoc
k32\.lib mpr\.lib winmm\.lib  version\.lib odbc32\.lib odbccp32\.lib msvcrt\.lib
&nbsp;    libc=msvcrt\.lib\, so=dll\, useshrplib=yes\, libperl=perl56\.lib
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_win32\.xs\, dlext=dll\, d\_dlsymun=undef\, ccdlflags=' '
&nbsp;    cccdlflags=' '\, lddlflags='\-dll \-nologo \-nodefaultlib \-release  \-libpath&#8203;:"C&#8203;:
\\Perl\\lib\\CORE"  \-machine&#8203;:x86'


Characteristics of this binary \(from libperl\)&#8203;:
&nbsp;  Compile\-time options&#8203;: MULTIPLICITY USE\_ITHREADS PERL\_IMPLICIT\_CONTEXT PERL\_IMP
LICIT\_SYS
&nbsp;  Locally applied patches&#8203;:
&nbsp;        ActivePerl Build 623
&nbsp;  Built under MSWin32
&nbsp;  Compiled at Dec 15 2000 16&#8203;:27&#8203;:07
&nbsp;  @&#8203;INC&#8203;:
&nbsp;    C&#8203;:/Perl/lib
&nbsp;    C&#8203;:/Perl/site/lib
&nbsp;    \.

\-\- 
Where there's a will\, there's a lawyer\.

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

\[\.\.\.\]

 > but it works under bleadperl
 > 
 > $ bleadperl \-wle '$foo = \\\-f "blah";  print $foo;  print $$foo'
 > SCALAR\(0x101947d8\)
 > Use of uninitialized value in print at \-e line 1\.
 > 
 > $ bleadperl \-wle '$foo = \\\-f "Mailbox";  print $foo;  print $$foo'
 > SCALAR\(0x10194cb0\)
 > 1
 > 
 > 
 > Grab the latest bleadperl snapshot and see if everything works ok on
 > your system & prefered config\.
 > ftp&#8203;://ftp\.funet\.fi/pub/languages/perl/snap/

It was fixed by changes \#12811 and \#12812\.

 > Meanwhile\, here's a test patch&#8203;:
 > 
 > \-\-\- t/run/kill\_perl\.t    2001/11/27 20&#8203;:20&#8203;:58    1\.1
 > \+\+\+ t/run/kill\_perl\.t    2001/11/27 20&#8203;:29&#8203;:09
 > @&#8203;@&#8203; \-809\,3 \+809\,9 @&#8203;@&#8203;
 >  EXPECT
 >  Can't modify constant item in read at \- line 1\, near "1\)"
 >  Execution of \- aborted due to compilation errors\.
 > \+\#\#\#\#\#\#\#\# scalar ref to file test operator segfaults on 5\.6\.1 \[ID 20011127\.155\]
 > \+\# This only happens if the filename is 11 characters or less\.
 > \+$foo = \\\-f "blah";
 > \+print "ok" if ref $foo && \!$$foo;
 > \+EXPECT
 > \+ok

May not be needed as it is already tested in t/op/stat\.t\.


Thanks\,
Radu Greab
